espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -319,7 +329,21 @@ def recognize(self, x, recog_args, char_list=None, rnnlm=None, use_jit=False):
         :rtype: list
         """
         enc_output = self.encode(x).unsqueeze(0)
-        if recog_args.ctc_weight > 0.0:
+        if self.mtlalpha == 1.0:
+            recog_args.ctc_weight = 1.0
+            logging.info("Set to pure CTC decoding mode.")
+
+        if self.mtlalpha > 0 and recog_args.ctc_weight == 1.0:
+            from itertools import groupby
+
+            lpz = self.ctc.argmax(enc_output)
+            collapsed_indices = [x[0] for x in groupby(lpz[0])]
+            hyp = [x for x in filter(lambda x: x != self.blank, collapsed_indices)]
+            nbest_hyps = [{"score": 0.0, "yseq": hyp}]
+            # NOTE: Curently, greedy decoding is supported only.
+            # TODO(hirofumi0810): Implement beam search
```

Comment:
It might be safer to use logging error or raise NotImplementedError if beam size > 1.